test/parallel/test-stream-pipe-error-unhandled.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { Readable, Writable } = require('stream');
+
+process.on('uncaughtException', common.mustCall((err) => {
+  assert.strictEqual(err.message, 'asd');
+}));
+
+const r = new Readable({
+  read() {
+    this.push('asd');
+  }
+});
+const w = new Writable({
+  autoDestroy: true,
+  write() {}
+});
+
+r.pipe(w);
+w.destroy(new Error('asd'));
```

Comment:
hmm... this ends up changing the semantics of `destroy()` rather significantly in that prior to this change, subsequent `destroy()` calls would be non-ops. This is not necessarily a bad thing in that I tend to prefer the notion that most calls on stream objects should fail with an error after `destroy()` is called. What I'm not certain about, however, is whether this should be direct throw or an emitted throw.

Reply:
I'm a little confused. The only change as far I can tell is that the emitted error will be propagated to unhandledException?

Reply:
I'm not sure I follow. All this PR does is to re-emit the already emitted error.

Reply:
I think in other places we just re-emit (like this) instead of throwing, e.g. https://github.com/nodejs/node/blob/master/lib/internal/streams/legacy.js#L56